src/components/TabPanel/index.tsx

```diff
@@ -12,25 +17,39 @@ export default function TabPanel(props: Props) {
     const {
         name,
         elementRef,
+        activeClassName,
+        retainMount = false,
+        className,
         ...otherProps
     } = props;
 
     let isActive = false;
+    const [isLoadedOnce, setIsLoadedOnce] = useState(false);
 
     if (context.useHash) {
         isActive = context.hash === name;
     } else {
         isActive = context.activeTab === name;
     }
 
-    if (!isActive) {
+    useEffect(() => {
+        setIsLoadedOnce(retainMount ?? false);
+    }, [isActive, retainMount]);
```

Comment:
We should set isLoadedOnce when isActive is true using useEffect.

And use `(isLoadedOnce && retainMount) || isActive` logic